core/src/main/java/com/linecorp/armeria/common/DefaultFlagsProvider.java

```diff
@@ -0,0 +1,322 @@
+/*
+ *  Copyright 2022 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+package com.linecorp.armeria.common;
+
+import java.io.File;
+import java.net.InetAddress;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.util.Sampler;
+import com.linecorp.armeria.common.util.TransportType;
+import com.linecorp.armeria.server.TransientServiceOption;
+
+/**
+ * Implementation of {@link FlagsProvider} which provides default values to {@link Flags}.
+ */
+public final class DefaultFlagsProvider implements FlagsProvider {
```

Comment:
Could remove `public`?
```suggestion
final class DefaultFlagsProvider implements FlagsProvider {
```

Reply:
Should we add a private constructor and make it singleton?